pytest_container/runtime.py

```diff
@@ -314,21 +292,34 @@ def _get_image_entrypoint_cmd(
         defined.
 
         """
-        entrypoint = (
-            check_output(
-                [
-                    self.runner_binary,
-                    "inspect",
-                    "-f",
-                    f"{{{{.Config.{query_type}}}}}",
-                    image_url_or_id,
-                ]
-            )
-            .decode("utf-8")
-            .strip()
+        entrypoint = self.run_command(
+            "inspect", "-f", f"{{{{.Config.{query_type}}}}}", image_url_or_id
         )
         return None if entrypoint == "[]" else entrypoint
 
+    def run_command(
+        self,
+        *args: str,
+        ignore_errors: bool = False,
+        strip: bool = True,
+    ) -> str:
+        cmd = [self.runner_binary]
+        cmd.extend(args)
+        result: str
+        try:
+            # universal_newlines is an alias for text before python 3.7
+            result = subprocess.check_output(
+                cmd, stderr=subprocess.PIPE, universal_newlines=True
+            )
+        except subprocess.CalledProcessError as exc:
+            if not ignore_errors:
+                _logger.error(exc.stderr)
+                raise exc
+            result = exc.stdout
```

Comment:
Why not use `subprocess.run()` instead?
Also, I would, highly, highly suggest to log the call. Testinfra does this and it is a savior knowing which commands are run.

Reply:
I'll change it to {{run()}} and add logging. Note that the current code doesn't log any of these either.

Reply:
The current code does in indeed not log the call, but since this now becomes the replacement of `testinfra`'s `check_output`, which logs the output & result, I think it should log too.